googleapiclient/http.py

```diff
@@ -661,37 +661,32 @@ def __init__(
 
 
 class MediaIoBaseDownload(object):
-    """ "Download media resources.
+    """Download media resources.
 
     Note that the Python file object is compatible with io.Base and can be used
     with this class also.
 
-
     Example:
-      request = farms.animals().get_media(id='cow')
-      fh = io.FileIO('cow.png', mode='wb')
-      downloader = MediaIoBaseDownload(fh, request, chunksize=1024*1024)
-
-      done = False
-      while done is False:
-        status, done = downloader.next_chunk()
-        if status:
-          print "Download %d%%." % int(status.progress() * 100)
-      print "Download Complete!"
+        request = farms.animals().get_media(id='cow')
+        downloader = MediaIoBaseDownload(request, chunksize=1024*1024)
+
+        for chunk, status, done in downloader.next_chunk():
+            with open('cow.png', 'ab') as cow_file:
+                cow_file.write(chunk)
+            print("Download %d%%." % int(status.progress() * 100))
+
+        print("Download Complete!")
     """
 
-    @util.positional(3)
-    def __init__(self, fd, request, chunksize=DEFAULT_CHUNK_SIZE):
+    @util.positional(2)
+    def __init__(self, request, chunksize=DEFAULT_CHUNK_SIZE):
         """Constructor.
 
         Args:
-          fd: io.Base or file object, The stream in which to write the downloaded
```

Comment:
To avoid breaking existing users, ideally we should continue support the existing behaviour as well as the new desired behaviour.